direct/core/src/main/java/cz/o2/proxima/direct/core/BulkAttributeWriter.java

```diff
@@ -52,4 +52,11 @@ default Type getType() {
    * @param statusCallback callback to commit the data
    */
   void write(StreamElement data, long watermark, CommitCallback statusCallback);
+
+  /**
+   * Update watermark when no input data arrives.
```

Comment:
I think `when no input data arr...` is just the impl. detail. Should we change this to `Update watermark of this writer.` instead.

Reply:
I don't think this is implementation detail, as the watermark is passed to `write` method. This is called *solely* when there is no input data currently.